src/colorizer/utils/data_load_utils.ts

```diff
@@ -87,3 +88,64 @@ export async function loadFromParquetUrl<T extends FeatureDataType>(url: string,
   });
   return { data, min: dataMin, max: dataMax };
 }
+
+/**
+ * Parses a ZIP file into a map of file paths to their contents. If files are nested
+ * inside one or more levels of empty directories, those directories will be stripped
+ * from the file paths.
+ * @param zipFile A ZIP file object.
+ * @returns A Promise that resolves to a map of file paths to a File object.
+ */
+export async function zipToFileMap(zipFile: File): Promise<Record<string, File>> {
+  const zip = await JSZip.loadAsync(zipFile);
+
+  // Load all contents and save them as File objects
+  const fileMap: Record<string, File> = {};
+  const filePromises: Promise<void>[] = [];
+  const loadToFileMap = async (relativePath: string, zipObject: JSZip.JSZipObject): Promise<void> => {
+    const fileContents = await zipObject.async("blob");
+    fileMap[relativePath] = new File([fileContents], relativePath);
+  };
+
+  zip.forEach((relativePath, zipObject) => {
+    if (zipObject.dir) {
+      return;
```

Comment:
So will this just skip over any files that are in a directory within the zip file?